lib/rules/badge-missing-aria-v9.js

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+"use strict";
+
+const { hasAssociatedLabelViaAriaLabelledBy } = require("../util/labelUtils");
+const { hasNonEmptyProp } = require("../util/hasNonEmptyProp");
+var elementType = require("jsx-ast-utils").elementType;
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+    meta: {
+        // possible error messages for the rule
+        messages: {
+            missingLabelOnBadge: "Badge need accessible labelling: aria-label or aria-labelledby"
+        },
+        // "problem" means the rule is identifying code that either will cause an error or may cause a confusing behavior: https://eslint.org/docs/latest/developer-guide/working-with-rules
+        type: "problem",
+        // docs for the rule
+        docs: {
+            description: "Accessibility: Badge need accessible labelling: aria-label or aria-labelledby",
+            recommended: true
+        },
+        schema: []
+    },
+
+    create(context) {
+        return {
+            // visitor functions for different types of nodes
+            JSXOpeningElement(node) {
+                // if it is not a Badge, return
+                if (elementType(node) !== "Badge") {
```

Comment:
I'm not sure if we need two separate rules for the Badge component.

A badge will take the accessible name of the parent such as a Tooltip or a Button, and will need to further labelling in that case.

A badge containing text content will have an accessible name so there is no need for other labels.

If a badge has an icon, we must add aria-label or aria-labelledby UNLESS the Badge is marked as decorative with aria-hidden.

If a badge has no content or an icon and the color conveys meaning, we can add the meaning with an aria-label. It depends on the context.


Suggestion:
We can consolidate the two newly created rules into one and we can have code like this:


```
                // if it has a tooltip parent, return
                if (hasToolTipParent(context)) {
                    return;
                }

                // if it has text content, return
                if (hasTextContentChild(node)) {
                    return;
                }

                // the button has an associated label
                if (hasAssociatedLabelViaAriaLabelledBy(openingElement, context)) {
                    return;
                }
```